module/zfs/zfs_ioctl.c

```diff
@@ -5037,6 +5038,16 @@ zfs_ioc_pool_reopen(zfs_cmd_t *zc)
 {
 	spa_t *spa;
 	int error;
+	nvlist_t *reopenflags;
+	boolean_t scrub_restart = B_TRUE;
+
+	if (zc->zc_nvlist_src != 0 &&
+	    (error = get_nvlist(zc->zc_nvlist_src, zc->zc_nvlist_src_size,
+	    zc->zc_iflags, &reopenflags)) == 0) {
+		nvlist_lookup_boolean_value(reopenflags,
+		    "scrub_restart", &scrub_restart);
+		nvlist_free(reopenflags);
+	}
```

Comment:
Good thought, since we're introducing the `innvl` this would be the best time to do the migration.  

@ab-oe switching to new style ioctl is pretty straight forward.  You'll need to register `ZFS_IOC_POOL_REOPEN` with `zfs_ioctl_register` instead of `zfs_ioctl_register_pool` and move it up before the "legacy function signature" comment.  You'll also need to update to the new  style prototype for the `zfs_ioc_pool_reopen`.  The big change is that you'll be provided and `innvl` and  `outnvl` and won't need to handle unpacking them yourself.  The updated ioctl will be compatible as long as you update this function such that it allows for `innvl == NULL`.

Reply:
Ok, I'll change this.